smarts/core/vehicle.py

```diff
@@ -431,6 +432,10 @@ def attach_sensors_to_vehicle(sim, vehicle, agent_interface, mission_planner):
             )
 
         if agent_interface.drivable_area_grid_map:
+            if not sim.renderer:
+                raise Exception(
+                    "A renderer is required to add a drivable_area_grid_map.  Ensure that `renderer_optional=False` is set when creating SMARTS instance."
+                )
```

Comment:
All of the sensors use the same exception. Maybe this is a case for a new exception type?

Reply:
Good call.  Will add.